### PR TITLE
KEYCLOAK-8522: Migrate IdpHint tests. Remove unneeded tests.

### PR DESCRIPTION
**Pull request by ssilvert**

### CODE REVIEW

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java [hunk 1]
```diff
@@ -272,37 +272,6 @@ public void testFixDuplicationsByReviewProfile() {
         assertFederatedUser("testing-user", "testing-user@redhat.com", "pedroigor");
     }
 
-    /**
-     * Tests that duplication is detected and user wants to link federatedIdentity with existing account. He will confirm link by email
-     */
-    @Test
-    public void testLinkAccountByEmailVerification() throws Exception {
```
Comment (pedroigor) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java]: Removed in favor of org.keycloak.testsuite.broker.AbstractBrokerTest#testLinkAccountWithEmailVerified.

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java [hunk 2]
```diff
@@ -355,61 +324,6 @@ public void testLinkAccountByEmailVerificationTwice() throws Exception {
         Assert.assertThat(infoPage.getInfo(), startsWith("You successfully verified your email. Please go back to your original browser and continue there with the login."));
     }
 
-    /**
-     * Tests that duplication is detected and user wants to link federatedIdentity with existing account. He will confirm link by email
-     */
-    @Test
-    public void testLinkAccountByEmailVerificationDifferentBrowser() throws Exception, Throwable {
```
Comment (pedroigor) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java]: Removed in favor of org.keycloak.testsuite.broker.AbstractBrokerTest#testVerifyEmailInNewBrowserWithPreserveClient.

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/IdentityProviderHintTest.java [hunk 3]
```diff
@@ -1,139 +0,0 @@
...
-import org.keycloak.testsuite.rule.WebResource;
-import org.keycloak.testsuite.rule.WebRule;
-import org.openqa.selenium.WebDriver;
-
-import static org.junit.Assert.assertTrue;
-
-/**
- * @author pedroigor
- */
-public class IdentityProviderHintTest {
```
Comment (pedroigor) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/IdentityProviderHintTest.java]: Removed in favor of new hint test added in the PR.

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/IdentityProviderRegistrationTest.java [hunk 4]
```diff
@@ -1,106 +0,0 @@
...
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-
-/**
- * @author pedroigor
- */
-public class IdentityProviderRegistrationTest extends AbstractIdentityProviderModelTest {
```
Comment (pedroigor) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/IdentityProviderRegistrationTest.java]: Removed in favor of existing tests that already check SPI functionality.

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/ImportIdentityProviderTest.java [hunk 5]
```diff
@@ -1,365 +0,0 @@
...
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-
-/**
- * @author pedroigor
- */
-public class ImportIdentityProviderTest extends AbstractIdentityProviderModelTest {
```
Comment (pedroigor) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/ImportIdentityProviderTest.java]: Removed in favor of existing tests that already check export/import realms with brokers.

**Review by pedroigor — DISMISSED** [review]

File: testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractKeycloakIdentityProviderTest.java [hunk 6]
```diff
@@ -97,46 +97,6 @@ public void testDisabledUser() throws Exception {
         }
     }
 
-    @Test
-    public void testTemporarilyDisabledUser() {
-        KeycloakSession session = brokerServerRule.startSession();
-        setUpdateProfileFirstLogin(session.realms().getRealmByName("realm-with-broker"), IdentityProviderRepresentation.UPFLM_OFF);
```
Comment (ssilvert) [inline testsuite/integration-deprecated/src/test/java/org/keycloak/testsuite/broker/AbstractKeycloakIdentityProviderTest.java]: Removed in favor of AbstractBrokerTest#loginWithExistingUserWithBruteForceEnabled

**Comment by pedroigor** [comment]
@keycloak-ci-bot test

**Comment by hmlnarik** [comment]
@keycloak-ci-bot test

**Comment by pedroigor** [comment]
@keycloak-ci-bot test

**Comment by ssilvert** [comment]
@keycloak-ci-bot test

**Review by pedroigor — APPROVED** [review]

**Pull request merged**